### PR TITLE
feat: open remote folders

### PR DESCRIPTION
**Pull request by brstuff**
feature/vscode-remote

Features:
- Open remote folders 
- Open remote files and workspaces 
- Search case-insensitive
- Rename "Project.code-workspace" to "Project Workspace"

### CODE REVIEW

**Comment by MrMarble** [comment]
Wow thank you for the contribution!

File: src/provider.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export default class VSCodeSearchProvider implements AppSearchProvider {
         continue
       }
       const name = path.split("/").pop()!;
-      this.workspaces[uniqueId()] = { name, path: path.replace("file://", "") };
+      this.workspaces[uniqueId()] = { name: name.replace(".code-workspace", " Workspace") , path: path.replace("file://", "") };
```
Comment (MrMarble) [inline src/provider.ts:42]: Why this change? Is this added to remote workspaces? I don't see it on my results so I'm curious

File: src/provider.ts [hunk 2]
```diff
@@ -94,10 +94,19 @@ export default class VSCodeSearchProvider implements AppSearchProvider {
...
   activateResult(result: string): void {
     if (this.app) {
-      this.app?.app_info.launch(
-        [Gio.file_new_for_path(this.workspaces[result].path)],
-        null,
-      );
+      const path = this.workspaces[result].path;
+      if (path.startsWith("vscode-remote://")) {
+        const lastSegment = path.split('/').pop();
+        const type = lastSegment?.includes('.') ? 'file' : 'folder';
```
Comment (MrMarble) [inline src/provider.ts]: I think this check could be problematic, specially on linux where the `.` is used to hide files and folders, if the user open `.ssh` or `.config` or a personal folder like my [dotfiles](https://github.com/MrMarble/dotfiles) that are in a `.dotfiles` folder.  

Since its a remote path we can't check the type but we could ignore the first character to reduce false positives
```suggestion
        const type = lastSegment?.slice(1)?.includes('.') ? 'file' : 'folder';
```

**Review by MrMarble — CHANGES_REQUESTED** [review]

File: src/provider.ts (hunk 1 above)
Comment (brstuff) [reply src/provider.ts:42]: In my case, I use a lot of remote workspaces. These are files with ".code-workspace" extension, what remains in the JSON is the filename "/path/to/project.code-workspace" file, so what appears in the list will be "project.code-workspace". The ".code-workspace" extension only adds noise. So I changed it to "projet Workspace" to make it cleaner and more understandable.
It's a file but in the end it ends up being a folder (workspace)

I don't think anyone will find it bad, but if that happens, we can create a setting to toggle the change the replace on and off.

File: src/provider.ts (hunk 2 above)
Comment (brstuff) [reply src/provider.ts]: Perfect

**Comment by MrMarble** [comment]
I'll merge your changes and upload the extension for revision, once is reviewed I'll create a new release
